src/main/java/io/jenkins/plugins/conventionalcommits/utils/HelmProjectType.java

```diff
@@ -30,27 +48,53 @@ public boolean check(File directory) {
   /**
    * Return the next version of the version attribute.
    *
-   * @param directory The project's directory.
+   * @param directory     The project's directory.
    * @param processHelper Not used.
    * @return The next calculated version (based on Semver).
    * @throws IOException If an error occur reading files.
    */
   @Override
   public Version getCurrentVersion(File directory, ProcessHelper processHelper) throws IOException {
-    Objects.requireNonNull(directory);
-
-    ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
-    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-
-    // Convert Chart.yaml to a DTO
-    HelmChart chart =
-        mapper.readValue(
-            new File(directory.getAbsoluteFile() + File.separator + CHART_YAML_NAME),
-            HelmChart.class);
-    return Version.valueOf(chart.getVersion());
+    // readChartFile(directory) can't be null, an Exception is thrown if a problem occur
+    // (file note found, file malformed, ..)
+    return Version.valueOf(readChartFile(directory).getVersion());
   }
 
+  /**
+   * Write back to the Chart.yaml the new Version.
+   *
+   * @param directory     The directory where is the Chart.yaml. <b>Mandatory</b>
+   * @param nextVersion   The next version to write. <b>Mandatory</b>
+   * @param processHelper Can be null, not used.
+   * @throws IOException          If an error occurred when accessing files or directory.
+   * @throws InterruptedException Not used.
+   */
   @Override
   public void writeVersion(File directory, Version nextVersion, ProcessHelper processHelper)
-      throws IOException, InterruptedException {}
+      throws IOException, InterruptedException {
+    Objects.requireNonNull(nextVersion);
+
+    HelmChart helmChart = readChartFile(directory);
+    helmChart.setVersion(nextVersion.toString());
+
+    yamlMapper.writeValue(new File(directory.getAbsoluteFile() + File.separator + CHART_YAML_NAME),
+        helmChart);
```

Comment:
thanks I'll take a look :wink:

Reply:
you're right, my  unit test test only the update of the version field, but my DTO have only this field so when I serialized it the result file have only the version field :sweat_smile: ... my fault.

I change the way to update the chart.yaml (and the unit test so that it  failed with the actual implementation) and I ask you to review again when it's done.

Reply:
No problem. I am grateful I could help :)